.github/actions/generate-sbom/docker-scan.yml

```diff
@@ -0,0 +1,46 @@
+name: "Docker Scan and SBOM"
+description: "Scan docker containers and generate SBOM"
+
+inputs:
+  docker_image:
+    description: "The docker image and tag to be scanned"
+    required: true
+
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install Trivy
+      run: |
+        curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v0.34.0
+
+    - name: run trivy docker
+        run: |
+          trivy image \
+          --format github \
+          --vuln-type os,library \
+          --security-checks vuln \
+          --output dependency-results.sbom.json ${{ inputs.docker_image }}
```

Comment:
nit: could you please indent the args like so.  It doesn't change behaviour at all - just makes it easier to read:
```sh
trivy image \
    --format github \
    --vuln-type os,library \
    --security-checks vuln \
    --output dependency-results.sbom.json ${{ inputs.docker_image }}
```